src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java

```diff
@@ -166,15 +167,13 @@ public ResponseEntity<VisitDto> addVisitToOwner(Integer ownerId, Integer petId,
     @Override
     public ResponseEntity<PetDto> getOwnersPet(Integer ownerId, Integer petId) {
         Owner owner = this.clinicService.findOwnerById(ownerId);
-        Pet pet = this.clinicService.findPetById(petId);
-        if (owner == null || pet == null) {
+        if (owner == null) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        } else {
-            if (!pet.getOwner().equals(owner)) {
-                return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-            } else {
-                return new ResponseEntity<>(petMapper.toPetDto(pet), HttpStatus.OK);
-            }
         }
+        Optional<Pet> pet = owner.getPets().stream().filter(p -> p.getId().equals(petId)).findFirst();
```

Comment:
IMO it's better to let logic goes in Service class, so controllers are focused on endpoint input/output. So i suggest to do : 
```java
Optional<Pet> target = this.clinicService.getPetByOwner(owner, petId);
```

Reply:
A better place is the `Owner` class ;)

Reply:
alright